src/styles/main.css

```diff
@@ -0,0 +1,98 @@
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  counter-reset: day-number 0;
+  display: flex;
+  flex-wrap: wrap;
+  width: calc(100px * 7 + 1px * 6 + 10px * 2);
+  padding: 10px;
+  gap: 1px;
+}
+
+.calendar__day {
+  width: 100px;
+  height: 100px;
+  background-color: #eee;
+  border: 1px solid #000;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+}
+
+.calendar__day::before {
+  content: counter(day-number);
+  counter-increment: day-number;
+  font-size: 30px;
+  font-family: Arial, sans-serif;
+  color: #000;
+  position: absolute;
+}
+
+.calendar__day:hover {
+  background-color: #ffbfcb;
+  cursor: pointer;
+  transform: translateY(-20px);
+  transition:
+    transform 0.5s,
+    background-color 0.5s;
+}
+
+.calendar--start-day-mon .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 0);
+}
+
+.calendar--start-day-tue .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 1);
+}
+
+.calendar--start-day-wed .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 2);
+}
+
+.calendar--start-day-thu .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 3);
+}
+
+.calendar--start-day-fri .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 4);
+}
+
+.calendar--start-day-sat .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 5);
+}
+
+.calendar--start-day-sun .calendar__day:nth-child(1) {
+  margin-left: calc((100px + 1px) * 6);
+}
```

Comment:
These styles can be generated using a loop in SCSS, which will make your code much cleaner and easier to maintain.